include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -6324,9 +6329,8 @@ namespace VectorTools
               // take the trace of the derivatives, square it, multiply it
               // with the weight function
               for (unsigned int k=0; k<dim; ++k)
-                sum += (data.psi_grads[q][k][k] * data.psi_grads[q][k][k]) *
-                       data.weight_vectors[q](k);
-              diff += sum * fe_values.JxW(q);
+                sum += data.psi_grads[q][k][k] * data.weight_vectors[q](k);
+              diff += sum * sum * fe_values.JxW(q);
```

Comment:
Good question. From the documentation of the integrate_difference function it is not clear to me if it is a weight for the norm (that is what we do for L2) or the function to be integrated. For Hdiv we evaluate the L2 norm of a scalar function (div u) so supplying a weight vector doesn't make much sense if it is a weight for the norm.

It makes more sense to me if weight would be defined to scale the components of the input function, but I guess it is too late to change this.

Should I change it to sqrt(weight) here?